repos/system_upgrade/el7toel8/actors/nischeck/libraries/nischeck.py

```diff
@@ -0,0 +1,166 @@
+import re
+import ipaddress
+
+from leapp import reporting
+
+
+class NisScanner:
+    """
+    Check configuration of /etc/yp.conf and /etc/nsswitch.conf. Address of NIS server cannot be
+    specified by host name if NIS is used to resolve host names.
+    Parameters:
+        sources (set) set of labels from nsswitch
+        data (dict) list of values for each source
+    """
+
+    def scan(self):
+        self.scan_nsswitch()
+        hostnames = self.hostnames_in_yp_conf()
+
+        if "hosts" in self.data:
+            if "nis" in self.data["hosts"] and hostnames:
+                self.report_error(hostnames)
+            else:
+                self.report_successful()
+        else:
+            self.report_successful()
+
+    def report_error(self, hostnames):
+        reporting.create_report([
+            reporting.Title('Unsupported NIS configuration found'),
+            reporting.Summary("NIS may be used for domain name resolution only if NIS "
+                              "server is specified by IP. NIS servers specified by "
+                              "host name: {}".format(", ".join(hostnames))),
+            reporting.Severity(reporting.Severity.MEDIUM)
+        ])
+
+    def report_successful(self):
+        reporting.create_report([
+            reporting.Title('Checking NIS configuration'),
+            reporting.Summary('Check successful'),
+            reporting.Severity(reporting.Severity.INFO)
+        ])
+
+    def hostnames_in_yp_conf(self):
+        """
+        Scan /etc/yp.conf, return if file contains domain name
+        """
+
+        path = "/etc/yp.conf"
+        try:
+            with open(path) as f:
+                lines = f.readlines()
+        except IOError:
+            return False
+
+        # remove comments
+        lines = list(map(lambda x: x.split("#", 1)[0], lines))
```

Comment:
You don't need to create list from the mapping. Map produces iterator as well

Reply:
Try to avoid `materializing` iterators, so if you really need `lines` to only iterate it once, then you don't need to create list from it